### DISCUSSION TITLE
部署时发生错误

### DISCUSSION THREAD

**Discussion by AlwsLvin** [pengxiao-song/LaWGPT#92] [Q&A（问答）]
以下是错误代码，折腾了很久了，尝试了网上许多方法也无法解决，希望有大佬帮忙解决。
===================================BUG REPORT===================================
Welcome to bitsandbytes. For bug reports, please run

python -m bitsandbytes

 and submit this information together with your error trace to: https://github.com/TimDettmers/bitsandbytes/issues
================================================================================
bin /home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cuda120.so
False
/home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: /home/linzechuan/anaconda3 did not contain ['libcudart.so', 'libcudart.so.11.0', 'libcudart.so.12.0'] as expected! Searching further paths...
  warn(msg)
/home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: WARNING: The following directories listed in your path were found to be non-existent: {PosixPath('/usr/local/cuda-11.1/lib64')}
  warn(msg)
/home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: /usr/local/cuda-11.1/lib64 did not contain ['libcudart.so', 'libcudart.so.11.0', 'libcudart.so.12.0'] as expected! Searching further paths...
  warn(msg)
/home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: WARNING: The following directories listed in your path were found to be non-existent: {PosixPath('@/tmp/.ICE-unix/2003,unix/lzc'), PosixPath('local/lzc')}
  warn(msg)
/home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: WARNING: The following directories listed in your path were found to be non-existent: {PosixPath('/etc/xdg/xdg-ubuntu')}
  warn(msg)
/home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: WARNING: The following directories listed in your path were found to be non-existent: {PosixPath('localhost,127.0.0.0/8,'), PosixPath('1')}
  warn(msg)
/home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: WARNING: The following directories listed in your path were found to be non-existent: {PosixPath('0'), PosixPath('1')}
  warn(msg)
/home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: WARNING: The following directories listed in your path were found to be non-existent: {PosixPath('/org/gnome/Terminal/screen/d59a42d5_0b01_4976_a85a_9e98e93ce432')}
  warn(msg)
/home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: WARNING: The following directories listed in your path were found to be non-existent: {PosixPath('http'), PosixPath('7890'), PosixPath('//127.0.0.1')}
  warn(msg)
/home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: WARNING: The following directories listed in your path were found to be non-existent: {PosixPath('path=/run/user/1000/bus,guid=ff57f114427bb11f138bddc8647caf66'), PosixPath('unix')}
  warn(msg)
CUDA_SETUP: WARNING! libcudart.so not found in any environmental path. Searching in backup paths...
CUDA SETUP: CUDA runtime path found: /usr/local/cuda/lib64/libcudart.so
CUDA SETUP: Highest compute capability among GPUs detected: 8.6
CUDA SETUP: Detected CUDA version 120
CUDA SETUP: Loading binary /home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cuda120.so...
libcublas.so.12: cannot open shared object file: No such file or directory
CUDA SETUP: Something unexpected happened. Please compile from source:
git clone git@github.com:TimDettmers/bitsandbytes.git
cd bitsandbytes
CUDA_VERSION=120
python setup.py install
Traceback (most recent call last):
  File "/media/linzechuan/Trash/lawgpt/LaWGPT-main/webui.py", line 8, in <module>
    from peft import PeftModel
  File "/home/linzechuan/anaconda3/lib/python3.10/site-packages/peft/__init__.py", line 22, in <module>
    from .mapping import MODEL_TYPE_TO_PEFT_MODEL_MAPPING, PEFT_TYPE_TO_CONFIG_MAPPING, get_peft_config, get_peft_model
  File "/home/linzechuan/anaconda3/lib/python3.10/site-packages/peft/mapping.py", line 16, in <module>
    from .peft_model import (
  File "/home/linzechuan/anaconda3/lib/python3.10/site-packages/peft/peft_model.py", line 31, in <module>
    from .tuners import LoraModel, PrefixEncoder, PromptEmbedding, PromptEncoder
  File "/home/linzechuan/anaconda3/lib/python3.10/site-packages/peft/tuners/__init__.py", line 20, in <module>
    from .lora import LoraConfig, LoraModel
  File "/home/linzechuan/anaconda3/lib/python3.10/site-packages/peft/tuners/lora.py", line 36, in <module>
    import bitsandbytes as bnb
  File "/home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/__init__.py", line 6, in <module>
    from . import cuda_setup, utils, research
  File "/home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/research/__init__.py", line 1, in <module>
    from . import nn
  File "/home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/research/nn/__init__.py", line 1, in <module>
    from .modules import LinearFP8Mixed, LinearFP8Global
  File "/home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/research/nn/modules.py", line 8, in <module>
    from bitsandbytes.optim import GlobalOptimManager
  File "/home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/optim/__init__.py", line 6, in <module>
    from bitsandbytes.cextension import COMPILED_WITH_CUDA
  File "/home/linzechuan/anaconda3/lib/python3.10/site-packages/bitsandbytes/cextension.py", line 20, in <module>
    raise RuntimeError('''
RuntimeError: 
        CUDA Setup failed despite GPU being available. Please run the following command to get more information:

        python -m bitsandbytes

        Inspect the output of the command and see if you can locate CUDA libraries. You might need to add them
        to your LD_LIBRARY_PATH. If you suspect a bug, please take the information from python -m bitsandbytes
        and open an issue at: https://github.com/TimDettmers/bitsandbytes/issues

**Comment by wltgithub** [comment]
windows11 的安装也没成功.也是 cuda 错误，有没有高人指点一下

**Comment by Ch1ldman** [comment]
“scripts/webui.sh: line 5: python: command not found ”win11部署出现这个提示 有大佬指点下嘛？
